lib/module-utils.js

```diff
@@ -0,0 +1,171 @@
+        const unusedIdentifiers = new Set()
+        for (const [identifier, used] of exports) {
+          if (!used) unusedIdentifiers.add(identifier)
+        }
+        if (unusedIdentifiers.size !== 0) {
+          unusedImports.set(path, unusedIdentifiers)
+        }
+      }
+    }
+
+    return unusedImports
+  })
+}
+
+exports.load = function(path) {
+  return readFile(path, {encoding: 'utf8'}).then(content => exports.parse(path, content))
+}
+
+exports.parse = function(path, content) {
+  const result = {
+    path,
+    entry: false,
+    imports: new Map(),
+    exports: new Set()
+  }
+
+  const cli = new eslint.CLIEngine()
+  const options = cli.getConfigForFile(path)
+
+  if (!options.settings) options.settings = {}
+
+  if (options.settings.unusedModulesEntries) {
+    const patterns = options.settings.unusedModulesEntries
+    result.ignore = matchPatterns(path, patterns)
+  }
+
+  options.parserPath = options.parser
+  options.getFilename = () => path
+
+  const ast = parse(path, content, options)
+
+  let currentSource
+
+  const visitor = source => {
+    currentSource = source
+  }
+
+  const visitors = moduleVisitor.default(visitor, {
+    amd: true,
+    esmodule: true,
+    commonjs: true
+  })
+
+  visitors['ExportDefaultDeclaration'] = visitor
+
+  traverse(ast, node => {
+    if (visitors[node.type]) {
+      currentSource = null
+      visitors[node.type](node)
+
+      if (currentSource) {
+        switch (node.type) {
+          case 'ExportDefaultDeclaration': {
+            result.exports.add('default')
+            break
+          }
+          case 'ImportDeclaration':
+          case 'CallExpression': {
+            const resolvedPath = resolve(currentSource.value, options) || currentSource.value
+            const specifiers = result.imports.get(resolvedPath) || new Set()
+            result.imports.set(resolvedPath, specifiers)
+
+            if (node.type === 'ImportDeclaration') {
+              node.specifiers.forEach(specifier => {
+                if (specifier.type === 'ImportDefaultSpecifier') {
+                  specifiers.add('default')
+                } else if (specifier.type === 'ImportSpecifier') {
+                  specifiers.add(specifier.imported.name)
+                }
+              })
+            } else {
+              specifiers.clear()
+              specifiers.add('*')
+            }
+            break
+          }
+        }
+      }
+    }
+  })
+
+  return result
+}
+
+function traverse(node, visitor) {
+  if (node && typeof node === 'object' && 'type' in node) {
+    visitor(node)
+    for (const key in node) {
+      traverse(node[key], visitor)
+    }
+  } else if (Array.isArray(node)) {
+    for (const child of node) {
+      traverse(child, visitor)
+    }
+  }
+}
+
+function matchPatterns(str, patterns) {
+  for (const pattern of patterns) {
+    if (minimatch(str, pattern)) {
+      return true
+    }
+  }
+  return false
+}
+
+function readFile(path, options) {
```

Comment:
You can use `promisify` from `utils` to do `const readFile = promisify(fs.readFile)`

Reply:
Oh nice, I didn't realize that was builtin.